tests/reactable_test.jsx

```diff
@@ -2070,6 +2070,169 @@ describe('Reactable', function() {
+                            <Reactable.Td column='Description'>bacon</Reactable.Td>
+                            <Reactable.Td column='Tag'>new</Reactable.Td>
+                        </Reactable.Tr>
+                    </Reactable.Table>,
+                    ReactableTestUtils.testNode()
+                );
+            });
+
+            after(ReactableTestUtils.resetTestEnvironment);
+
+            context('from the filterer field', function() {
+                it('filters case insensitive on specified columns', function() {
+                    var $filter = $('#table thead tr.reactable-filterer input.reactable-filter-input');
+
+                    $filter.val('x');
+                    React.addons.TestUtils.Simulate.keyUp($filter[0]);
+
+                    ReactableTestUtils.expectRowText(0, ['New Mexico', 'lorem ipsum', 'old']);
+                    ReactableTestUtils.expectRowText(1, ['Colorado', 'lol', 'x']);
+                });
+
+                it('filter placeholder is set', function(){
+                    var $filter = $('#table thead tr.reactable-filterer input.reactable-filter-input');
+                    expect($filter.attr("placeholder")).to.equal('Filter Results');
+                })
+            });
+
+            context('from the function', function() {
+                before(function() {
+                    this.component.filterBy('l');
+                });
+
+                it('applies the filtering', function() {
+                    ReactableTestUtils.expectRowText(0, ['Colorado', 'lol', 'x']);
+                    ReactableTestUtils.expectRowText(1, ['Alaska', 'bacon', 'new']);
+                });
+
+                it('updates the value of the filterer', function() {
+                    var $filter = $('#table thead tr.reactable-filterer input.reactable-filter-input');
+                    expect($filter).to.have.value('l');
+                });
+            });
+
+            context('from filterBy prop', function() {
+                before(function() {
+                  ReactableTestUtils.resetTestEnvironment();
+
+                  var ParentComponent = React.createClass({
+                    getInitialState: function() {
+                      return {customFilterText: 'l'}
+                    },
+
+                    handleChange(event) {
+                      this.setState({customFilterText: event.target.value});
+                    },
+
+                    render: function() {
+                      return (
+                        <div>
+                          <input type="text" ref="customFilterInput" id="customFilterInput" value={this.state.customFilterText} onChange={this.handleChange}/>
+                          <Reactable.Table className="table" id="table"
+                              filterable={[
+                                  {
+                                      column: 'Tag',
+                                      filterFunction: function(contents) {
```

Comment:
My thought for this was that the `filterFunction` would be passed the datum *and the user input filter*, so that it could do stemming, case insensitive comparison, etc.